### PR TITLE
Implement JAX dispatch for Argsort and add `stable` argument to sorting functions

### PR DESCRIPTION
**Pull request by ricardoV94**
## Description

Also cleanup useless stuff from `tensor.sort` and add new `stable` kwarg for compat with numpy 2.0 signature.

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

File: pytensor/tensor/sort.py [hunk 1]
```diff
@@ -1,25 +1,32 @@
...
 from pytensor.tensor.basic import arange, as_tensor_variable, switch
-from pytensor.tensor.math import eq, ge, mul
+from pytensor.tensor.math import eq, ge
 from pytensor.tensor.type import TensorType
 
 
-def _variable_is_none(var):
-    return isinstance(var, Constant) and var.data is None
+KIND_VALUES = ["quicksort", "mergesort", "heapsort", "stable"]
+KIND = Literal["quicksort", "mergesort", "heapsort", "stable"]
```
Comment (jessegrabowski) [inline pytensor/tensor/sort.py:15]:
```suggestion
KIND = typing.get_args(KIND_VALUES)
```

This way there's no need to keep the two lists synchronized if they get changed in the future

File: pytensor/tensor/sort.py [hunk 2]
```diff
@@ -1,25 +1,32 @@
...
-def _check_tensor_is_scalar(var):
-    """
-    Checks if a tensor variable is scalar, raise ValueError otherwise
-    """
-    msg = "%(var)s is expected to be 0d tensor, got %(ndim)d"
-    if var.ndim != 0:
-        raise ValueError(msg % (var, var.ndim))
+def _parse_sort_args(kind: KIND | None, order, stable: bool | None) -> KIND:
+    if order is not None:
+        raise ValueError("order is not applicable to PyTensor graphs")
```
Comment (jessegrabowski) [inline pytensor/tensor/sort.py:20]:
```suggestion
        raise ValueError("The "order" argument not used by PyTensor and cannot be set. It is included for consistency with numpy only")
```

Bit more obvious what's going on

File: tests/tensor/test_sort.py [hunk 3]
```diff
@@ -208,14 +209,16 @@ def test_argsort():
...
-    axis = lscalar()
-    a1 = ArgSortOp("mergesort", [])
-    a2 = ArgSortOp("quicksort", [])
+    a1 = ArgSortOp(
+        "mergesort",
+    )
+    a2 = ArgSortOp(
+        "quicksort",
+    )
     # All the below should give true
```
Comment (jessegrabowski) [inline tests/tensor/test_sort.py:218]: Not your code, but this comment is extremely silly

**Review by jessegrabowski — APPROVED** [review]
I like that you're already moving stuff out of `basic` into specialized files, makes the repo nicer. Are there any rewrites related to sort that should be broken out into a separate file?

File: pytensor/tensor/sort.py (hunk 2 above)
Comment (ricardoV94) [reply pytensor/tensor/sort.py:20]: It is included only so we can raise this informative error :D

File: tests/tensor/test_sort.py (hunk 3 above)
Comment (ricardoV94) [reply tests/tensor/test_sort.py:218]: It is xD, I'll take it out

File: pytensor/tensor/sort.py (hunk 1 above)
Comment (ricardoV94) [reply pytensor/tensor/sort.py]: Your suggestion did not work, did you mean something slightly different?

Comment (jessegrabowski) [reply pytensor/tensor/sort.py]: Yes, sorry. Here's what I have in mind:

```
KIND= Literal["quicksort", "mergesort", "heapsort", "stable"]
KIND_VALUES: tuple[KIND, ...] = typing.get_args(KIND)
```

**Pull request merged**